project-icp/bee-pollinator-app
# Aggregate CDL values by category

#### Overview
Two sets of crop data were provided to us by the ICP Team: cdl_data.csv containing mean scores for forage and nesting suitability by crop type and the CDL raster of land use cover describing crop distribution spatially.  The CDL has over 100 distinct types, but they were grouped into approximately 46 categories.  Two scripts were created which aggregate both data sets on that grouping to produce a more concise data set.  Some output of the scripts:
* A mapping of a new id to a list of old ids 
* A CSV of cdl_data compressed into group values only
* An uncompressed CDL tif with values reclassified according to that aggregation
* A compressed and tiled CDL tif suitable as a data source for the pollinator model

The model itself was updated to reference the new csv and tif as well as some assumptions about what crop types to report on based on the new ids.  The colormap for the tif was reduced as well, but the choices are temporary.

Connects #81 

#### Testing
* Reprovision the worker machine `vagrant provision worker`
* `vagrant ssh worker` and `cd /opt/app/pollinator/src/pollinator/reclass`
* Run `python group_cdl_data.py` to generate the new csv
* Run `python reclassify_cdl.py` to generate the two new tifs

The new CSV is committed in this PR but the compressed tif can be copied to your synced folder (`/opt/icp-crop-data` on your guest) at which point you can test the model by running the app and modeling a small AoI.
* Rebundle the javascript `./scripts/bundle.sh --debug`
* Search for `Drummond Dr, Yuba City, CA 95991` which should get you to an area with two pollinator dependent crop types: Almond (33) and Orchard (27)
![screenshot from 2016-12-22 10 36 52](https://cloud.githubusercontent.com/assets/1014341/21430772/913c4fd2-c832-11e6-8143-febcaff300cb.png)

* Draw & confirm the output of the model conforms to the new grouped id scheme
```json
{
    "12": 0,
    "16": 0,
    "17": 0,
    "18": 0,
    "20": 0,
    "27": 0.8427295684814453,
    "33": 0.3804641366004944,
    "46": 0,
    "47": 0
}
```
